tests/rods.cpp

```diff
@@ -110,7 +110,7 @@ added_mass()
 				     << ": " << err << endl;
 				return false;
 			}
-			if (!isclose(pos[2], expected_z, 1e-8, 1e-10)) {
+			if (!isclose(pos[2], expected_z, 1e16, TOL)) {
```

Comment:
Was this test failing? The tolerances were pretty close, but it was passing on my system. The case is simple enough that the error is just integrator error. Additionally the last two parameters to isclose are relative tolerance and absolute tolerance, so I presume that the 1e16 relative tolerance is a typo (perhaps supposed to be 1e-16?). I don't have a specific objection to that change, but given that TOL is generally relatively loose, I slightly prefer it being set more strictly on this added mass test. But I'm also not sure there could be a reasonable change to MoorDyn that would cause added mass of rods to be incorrect that would cause this test to still pass so I'm not against this change. Also if the relative tolerance was meant to be 1e-16, then it could also just be set to zero, then isclose will behave like `isclose(a, b, atol):  abs(a-b) <= atol`.

Reply:
Yeah, it was failing to me:

`Node 0 of Rod 1 should have a z position of -9.98353 but has a z pos of -10 at t=0.1`

But it really seems to be broken. If I comment out [tests/rods.cpp:117](https://github.com/KelsonMarine/MoorDyn_Public/blob/v2/tests/rods.cpp#L117) I get this:

```
Node 0 of Rod 1 should have a z position of -9.98353 but has a z pos of -10 at t=0.1
Node 1 of Rod 1 should have a z position of -9.98353 but has a z pos of -10 at t=0.1
Node 2 of Rod 1 should have a z position of -9.98353 but has a z pos of -10 at t=0.1
Node 3 of Rod 1 should have a z position of -9.98353 but has a z pos of -10 at t=0.1
Node 0 of Rod 1 should have a z position of -9.93412 but has a z pos of -10 at t=0.2
Node 1 of Rod 1 should have a z position of -9.93412 but has a z pos of -10 at t=0.2
Node 2 of Rod 1 should have a z position of -9.93412 but has a z pos of -10 at t=0.2
Node 3 of Rod 1 should have a z position of -9.93412 but has a z pos of -10 at t=0.2
Node 0 of Rod 1 should have a z position of -9.85177 but has a z pos of -10 at t=0.3
Node 1 of Rod 1 should have a z position of -9.85177 but has a z pos of -10 at t=0.3
Node 2 of Rod 1 should have a z position of -9.85177 but has a z pos of -10 at t=0.3
Node 3 of Rod 1 should have a z position of -9.85177 but has a z pos of -10 at t=0.3
Node 0 of Rod 1 should have a z position of -9.73648 but has a z pos of -10 at t=0.4
Node 1 of Rod 1 should have a z position of -9.73648 but has a z pos of -10 at t=0.4
Node 2 of Rod 1 should have a z position of -9.73648 but has a z pos of -10 at t=0.4
Node 3 of Rod 1 should have a z position of -9.73648 but has a z pos of -10 at t=0.4
Node 0 of Rod 1 should have a z position of -9.58826 but has a z pos of -10 at t=0.5
Node 1 of Rod 1 should have a z position of -9.58826 but has a z pos of -10 at t=0.5
Node 2 of Rod 1 should have a z position of -9.58826 but has a z pos of -10 at t=0.5
Node 3 of Rod 1 should have a z position of -9.58826 but has a z pos of -10 at t=0.5
Node 0 of Rod 1 should have a z position of -9.40709 but has a z pos of -10 at t=0.6
Node 1 of Rod 1 should have a z position of -9.40709 but has a z pos of -10 at t=0.6
Node 2 of Rod 1 should have a z position of -9.40709 but has a z pos of -10 at t=0.6
Node 3 of Rod 1 should have a z position of -9.40709 but has a z pos of -10 at t=0.6
Node 0 of Rod 1 should have a z position of -9.19298 but has a z pos of -10 at t=0.7
Node 1 of Rod 1 should have a z position of -9.19298 but has a z pos of -10 at t=0.7
Node 2 of Rod 1 should have a z position of -9.19298 but has a z pos of -10 at t=0.7
Node 3 of Rod 1 should have a z position of -9.19298 but has a z pos of -10 at t=0.7
Node 0 of Rod 1 should have a z position of -8.94593 but has a z pos of -10 at t=0.8
Node 1 of Rod 1 should have a z position of -8.94593 but has a z pos of -10 at t=0.8
Node 2 of Rod 1 should have a z position of -8.94593 but has a z pos of -10 at t=0.8
Node 3 of Rod 1 should have a z position of -8.94593 but has a z pos of -10 at t=0.8
Node 0 of Rod 1 should have a z position of -8.66595 but has a z pos of -10 at t=0.9
Node 1 of Rod 1 should have a z position of -8.66595 but has a z pos of -10 at t=0.9
Node 2 of Rod 1 should have a z position of -8.66595 but has a z pos of -10 at t=0.9
Node 3 of Rod 1 should have a z position of -8.66595 but has a z pos of -10 at t=0.9
Node 0 of Rod 1 should have a z position of -8.35302 but has a z pos of -10 at t=1
Node 1 of Rod 1 should have a z position of -8.35302 but has a z pos of -10 at t=1
Node 2 of Rod 1 should have a z position of -8.35302 but has a z pos of -10 at t=1
Node 3 of Rod 1 should have a z position of -8.35302 but has a z pos of -10 at t=1
Node 0 of Rod 1 should have a z position of -8.00716 but has a z pos of -10 at t=1.1
Node 1 of Rod 1 should have a z position of -8.00716 but has a z pos of -10 at t=1.1
Node 2 of Rod 1 should have a z position of -8.00716 but has a z pos of -10 at t=1.1
Node 3 of Rod 1 should have a z position of -8.00716 but has a z pos of -10 at t=1.1
Node 0 of Rod 1 should have a z position of -7.62835 but has a z pos of -10 at t=1.2
Node 1 of Rod 1 should have a z position of -7.62835 but has a z pos of -10 at t=1.2
Node 2 of Rod 1 should have a z position of -7.62835 but has a z pos of -10 at t=1.2
Node 3 of Rod 1 should have a z position of -7.62835 but has a z pos of -10 at t=1.2
Node 0 of Rod 1 should have a z position of -7.21661 but has a z pos of -10 at t=1.3
Node 1 of Rod 1 should have a z position of -7.21661 but has a z pos of -10 at t=1.3
Node 2 of Rod 1 should have a z position of -7.21661 but has a z pos of -10 at t=1.3
Node 3 of Rod 1 should have a z position of -7.21661 but has a z pos of -10 at t=1.3
Node 0 of Rod 1 should have a z position of -6.77192 but has a z pos of -10 at t=1.4
Node 1 of Rod 1 should have a z position of -6.77192 but has a z pos of -10 at t=1.4
Node 2 of Rod 1 should have a z position of -6.77192 but has a z pos of -10 at t=1.4
Node 3 of Rod 1 should have a z position of -6.77192 but has a z pos of -10 at t=1.4
Node 0 of Rod 1 should have a z position of -6.2943 but has a z pos of -10 at t=1.5
Node 1 of Rod 1 should have a z position of -6.2943 but has a z pos of -10 at t=1.5
Node 2 of Rod 1 should have a z position of -6.2943 but has a z pos of -10 at t=1.5
Node 3 of Rod 1 should have a z position of -6.2943 but has a z pos of -10 at t=1.5
Node 0 of Rod 1 should have a z position of -5.78374 but has a z pos of -10 at t=1.6
Node 1 of Rod 1 should have a z position of -5.78374 but has a z pos of -10 at t=1.6
Node 2 of Rod 1 should have a z position of -5.78374 but has a z pos of -10 at t=1.6
Node 3 of Rod 1 should have a z position of -5.78374 but has a z pos of -10 at t=1.6
Node 0 of Rod 1 should have a z position of -5.24024 but has a z pos of -10 at t=1.7
Node 1 of Rod 1 should have a z position of -5.24024 but has a z pos of -10 at t=1.7
Node 2 of Rod 1 should have a z position of -5.24024 but has a z pos of -10 at t=1.7
Node 3 of Rod 1 should have a z position of -5.24024 but has a z pos of -10 at t=1.7
Node 0 of Rod 1 should have a z position of -4.66379 but has a z pos of -10 at t=1.8
Node 1 of Rod 1 should have a z position of -4.66379 but has a z pos of -10 at t=1.8
Node 2 of Rod 1 should have a z position of -4.66379 but has a z pos of -10 at t=1.8
Node 3 of Rod 1 should have a z position of -4.66379 but has a z pos of -10 at t=1.8
Node 0 of Rod 1 should have a z position of -4.05441 but has a z pos of -10 at t=1.9
Node 1 of Rod 1 should have a z position of -4.05441 but has a z pos of -10 at t=1.9
Node 2 of Rod 1 should have a z position of -4.05441 but has a z pos of -10 at t=1.9
Node 3 of Rod 1 should have a z position of -4.05441 but has a z pos of -10 at t=1.9
Node 0 of Rod 1 should have a z position of -3.41209 but has a z pos of -10 at t=2
Node 1 of Rod 1 should have a z position of -3.41209 but has a z pos of -10 at t=2
Node 2 of Rod 1 should have a z position of -3.41209 but has a z pos of -10 at t=2
Node 3 of Rod 1 should have a z position of -3.41209 but has a z pos of -10 at t=2
Node 0 of Rod 1 should have a z position of -2.73683 but has a z pos of -10 at t=2.1
Node 1 of Rod 1 should have a z position of -2.73683 but has a z pos of -10 at t=2.1
Node 2 of Rod 1 should have a z position of -2.73683 but has a z pos of -10 at t=2.1
Node 3 of Rod 1 should have a z position of -2.73683 but has a z pos of -10 at t=2.1
Node 0 of Rod 1 should have a z position of -2.02863 but has a z pos of -10 at t=2.2
Node 1 of Rod 1 should have a z position of -2.02863 but has a z pos of -10 at t=2.2
Node 2 of Rod 1 should have a z position of -2.02863 but has a z pos of -10 at t=2.2
Node 3 of Rod 1 should have a z position of -2.02863 but has a z pos of -10 at t=2.2
Node 0 of Rod 1 should have a z position of -1.28749 but has a z pos of -10 at t=2.3
Node 1 of Rod 1 should have a z position of -1.28749 but has a z pos of -10 at t=2.3
Node 2 of Rod 1 should have a z position of -1.28749 but has a z pos of -10 at t=2.3
Node 3 of Rod 1 should have a z position of -1.28749 but has a z pos of -10 at t=2.3
Node 0 of Rod 1 should have a z position of -0.51341 but has a z pos of -10 at t=2.4
Node 1 of Rod 1 should have a z position of -0.51341 but has a z pos of -10 at t=2.4
Node 2 of Rod 1 should have a z position of -0.51341 but has a z pos of -10 at t=2.4
Node 3 of Rod 1 should have a z position of -0.51341 but has a z pos of -10 at t=2.4
```

So it really seems the rod is not moving at all. For the time being I restore the old error tracking and further investigate what is going on